modules/tinymce/src/plugins/lists/main/ts/listmodel/ComposeList.ts

```diff
@@ -32,22 +32,30 @@ const createSegment = (scope: Document, listType: ListType): Segment => {
   return segment;
 };
 
-const createSegments = (scope: Document, entry: EntryList, size: number): Segment[] => {
+const createSegments = (scope: Document, entry: EntryList | EntryNoList, size: number): Segment[] => {
```

Comment:
There seems there is still a comment entry. I think this structure should be a discriminated union so have a kind of either `list-item` or `node` or maybe `fragment` if you combine all nodes that are non lists into an array of nodes.